go-controller/pkg/allocator/pod/pod_annotation.go

```diff
@@ -170,12 +173,18 @@ func allocatePodAnnotationWithTunnelID(
 	podAnnotation *util.PodAnnotation,
 	err error) {
 
+	node, err := kube.GetNode(pod.Spec.NodeName)
+	if err != nil {
+		return nil, nil, err
+	}
```

Comment:
We should not add an api call here as stated on the comment from original PR.

Reply:
ah missed it, so to do as you done with the factory ? (assuming it is reachable)

Reply:
Yep, like we pass the Node directly from controllers and we red the Node there.

Reply:
Done, ptal